js/gestionPresupuesto.js

```diff
@@ -1,18 +1,63 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
 
-// TODO: Variable global
 
 
-function actualizarPresupuesto() {
-    // TODO
+let presupuesto;
+presupuesto = 0;
+
+let gasto = {
+    descripcion : "",
+    valor : 0,
+
+
+    mostrarGasto() {
+        return `Gasto correspondiente a ${this.descripcion} con valor ${this.valor} €`;
+    },
+
+    actualizarDescripcion(descripcion) {
+    
+        this.descripcion = descripcion;
+    },
+
+    actualizarValor(valor) {
+    
+        if (valor >= 0) 
+        {
+            this.valor = valor;
+        }
+    }
+}
+
+
+
+
+function actualizarPresupuesto(valor) {
+    
+    if (valor >= 0) {
+    
+        presupuesto = valor;
+        return presupuesto;
+    } 
+    else {
+        console.log("La cantidad introducida no es valida");
+        return -1;
+    }
 }
 
 function mostrarPresupuesto() {
-    // TODO
+    return `Tu presupuesto actual es de ${presupuesto} €`;
 }
 
-function CrearGasto() {
-    // TODO
+function CrearGasto(descripcion, valor) {
+    gasto.descripcion = descripcion;
```

Comment:
No es correcto... Estás trabajando todo el rato con un objeto `gasto` que creas arriba.

Lo que tienes que hacer es utilizar la función `CrearGasto` como una función constructora y definir toda la lógica que has puesto en `gasto` (arriba del todo) mediante `this`: https://es.javascript.info/constructor-new#funcion-constructora

Si no, estás trabajando todo el rato sobre el mismo objeto. Si creas un segundo gasto sobreescribirás el primero.

Soluciónalo y abre otra PR cuando lo tengas.

Ánimo!!